llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4278,6 +4278,26 @@ VectorizationFactor LoopVectorizationPlanner::selectVectorizationFactor() {
           if (!VPI)
             continue;
           switch (VPI->getOpcode()) {
+          // Selects are not modelled in the legacy cost model if they are
+          // inserted for reductions.
```

Comment:
It took me a bit to understand the relation between reductions and divisions below, would it be a bit clearer to frame it something like "Selects are only modelled in the legacy cost model for safe divisors"?